src/megaapi_impl.cpp

```diff
@@ -3572,68 +3604,130 @@ MegaStringMapPrivate::MegaStringMapPrivate(const MegaStringMapPrivate *megaStrin
     delete keys;
 }
 
-MegaStringListPrivate::MegaStringListPrivate()
+
+MegaStringListPrivate::MegaStringListPrivate(char **newlist, int size)
+{
+    for (int i = 0; i < size; i++)
+    {
+        append(newlist[i]);
+    }
+}
+
+MegaStringList *MegaStringListPrivate::copy() const
 {
-    list = NULL;
-    s = 0;
+    auto list = new MegaStringListPrivate;
+    list->m_list = m_list;
+    return list;
 }
 
-MegaStringListPrivate::MegaStringListPrivate(MegaStringListPrivate *stringList)
+void MegaStringListPrivate::append(const char* value)
 {
-    s = stringList->size();
-    if (!s)
+    m_list.emplace_back(value);
+}
+
+const char *MegaStringListPrivate::get(int i) const
+{
+    if (i >= 0 && i < size())
     {
-        list = NULL;
-        return;
+        return m_list[i].c_str();
     }
+    return nullptr;
+}
 
-    list = new const char*[s];
-    for (int i = 0; i < s; i++)
-        list[i] = MegaApi::strdup(stringList->get(i));
+int MegaStringListPrivate::size() const
```

Comment:
I'm finding it very hard to review the changes to MegaStringListPrivate, as the diffs are all mixed up with the code for the  new classes, and also I think it's introducing an inefficiency where it used to just take ownership of some pointers, and now copies the content instead, affecting any call sites as well as the class itself.    I think we should leave this one alone - an "if it ain't broke" case

Reply:
Yes, you're right. I thought there was a bug but it probably is an issue in my test. Will change back...